pysat/_instrument.py

```diff
@@ -2632,18 +2585,24 @@ def load(self, yr=None, doy=None, end_yr=None, end_doy=None, date=None,
                 if (self.index[-1] == last_pad) & (not want_last_pad):
                     self.data = self[:-1]
 
-        # if self.pad is False, load single day
+        # If self.pad is False, load single day
         else:
             self.data, meta = self._load_data(date=self.date, fid=self._fid,
                                               inc=self.load_step)
             if not self.empty:
+                meta.accept_default_labels(self.meta)
```

Comment:
No, but also wasn't doing that in all cases.  I think it is actually cleaner to pass labels to load, and I intend to add that as a recommended optional kwarg in the docs, but that's something to discuss in a meeting.

Reply:
Integers in meta revealed a hole in `_load_data` that I will check out.

Reply:
Tests still pass locally for me without this statement. I also ran through the integer test manually and didn't run into an issue.

I printed out the metadata just after each of the current `accept_default_labels` calls but I don't see a difference.
```
Normal accept labels
pysat Meta object
-----------------
Tracking 12 metadata values
Metadata for 8 standard variables
Metadata for 0 ND variables

MetaLabels:
-----------
units->Units       name->Long_Name    notes->notes       
desc->desc         plot->plot         axis->axis         
scale->scale       min_val->value_min max_val->value_max 
fill_val->fill     

Standard Metadata variables:
uts       Epoch     mlt       
             ...              
longitude latitude  altitude  


second accept labels
pysat Meta object
-----------------
Tracking 12 metadata values
Metadata for 8 standard variables
Metadata for 0 ND variables

MetaLabels:
-----------
units->Units       name->Long_Name    notes->notes       
desc->desc         plot->plot         axis->axis         
scale->scale       min_val->value_min max_val->value_max 
fill_val->fill     

Standard Metadata variables:
uts       Epoch     mlt       
             ...              
longitude latitude  altitude  


```

Reply:
You need to print out the `repr` to see if `fill_val` is an int or not.  Of course, it's also possible I made a mistake.

Reply:
Will do. Thanks for the guidance.

Reply:
Here is what I'm getting both with and without this second `accept_default_labels`
```
In [1]: default_val = {'notes': '', 'units': '', 'desc': '', 
   ...:                             'scale': 'linear'}                                                                                                                                                                      

In [2]: default_name = ['long_name', 'axis', 'plot']                                                                                                                                                                        

In [3]: default_nan = ['fill', 'value_min', 'value_max']                                                                                                                                                                    

In [4]: import numpy as np                                                                                                                                                                                                  

In [5]: import pysat                                                                                                                                                                                                        

In [7]: meta_labels = {'units': ('Units', str), 
   ...:                             'name': ('Long_Name', str)}                                                                                                                                                             

In [8]:          meta_labels['fill_val'] = ("fill", int) 
   ...:          testInst = pysat.Instrument('pysat', 'testing', 
   ...:                                           clean_level='clean', 
   ...:                                           labels=meta_labels) 
   ...:          testInst.load(2009, 1) 
   ...:          meta = testInst.meta 
   ...:          dval = 'int32_dummy' 
   ...:  
   ...:           # Update the testing data 
   ...:          default_val['fill'] = -1 
   ...:          default_nan.pop(default_nan.index('fill')) 

In [9]:         for lkey in default_name: 
   ...:             assert meta[dval, lkey] == dval 
   ...:  
   ...:         for lkey in default_nan: 
   ...:             assert np.isnan(meta[dval, lkey]) 
   ...:  
   ...:         for lkey in default_val.keys(): 
   ...:             assert meta[dval, lkey] == default_val[lkey] 
   ...:  
   ...:         assert 'children' not in meta.data.columns 
   ...:         assert dval not in meta.keys_nD()                                                                                                                                                                           

In [11]: testInst.meta.data['fill']                                                                                                                                                                                         
Out[11]: 
uts              NaN
Epoch            NaN
mlt              NaN
slt              NaN
orbit_num        NaN
longitude        NaN
latitude         NaN
altitude         NaN
dummy1            -1
dummy2            -1
dummy3            -1
dummy4            -1
string_dummy      -1
unicode_dummy     -1
int8_dummy        -1
int16_dummy       -1
int32_dummy       -1
int64_dummy       -1
Name: fill, dtype: object
```

All of the integer data is getting -1 for the fill value. Floats are getting NaN (I haven't sorted it how it knows yet, traced a couple functions) but the strings are also getting -1. If the floats retain a NaN for fill despite a -1 default, shouldn't strings get a ''? 

I checked just a straight load (develop-3 branch) and got,
```
In [5]: test.meta.data['fill']                                                                                                                                                                                              
Out[5]: 
uts              NaN
Epoch            NaN
mlt              NaN
slt              NaN
orbit_num        NaN
longitude        NaN
latitude         NaN
altitude         NaN
dummy1           NaN
dummy2           NaN
dummy3           NaN
dummy4           NaN
string_dummy     NaN
unicode_dummy    NaN
int8_dummy       NaN
int16_dummy      NaN
int32_dummy      NaN
int64_dummy      NaN
Name: fill, dtype: object
```
This point about strings is a pre-existing condition and probably out of scope. If you have a solution though....

On point, I haven't found a difference with the second `accept_default_labels` and should be removed. Unit tests pass both ways.

Reply:
Ah, this is because meta default fill values don't take the variable type into account, only the label type into account.  That is definitely out of scope here and would change the way meta works.